### PR TITLE
[Wayland] updated all Wayland packages, dependencies & patches

### PR DESCRIPTION
**Pull request by 5schatten**
- This PR should allow to build `Kodi 18.2` without `X11` and use instead `Wayland/Weston` & `OpenGL ES`
- it updates all packages to the latest versions, added dependencies as `pugixml` & added or updated patches for them
- Generic build tested on an Intel system with these project changes:

```
  # OpenGL(X) implementation to use (no / mesa)
    OPENGL="no"

  # OpenGL-ES implementation to use (no / bcm2835-driver / gpu-viv-bin-mx6q)
    OPENGLES="mesa"

  # include uvesafb support (yes / no)
    UVESAFB_SUPPORT="yes"

  # Displayserver to use (x11 / no)
    DISPLAYSERVER="weston"

  # Xorg Graphic drivers to use (all / i915,i965,r200,r300,r600,nvidia)
  # Space separated list is supported,
  # e.g. GRAPHIC_DRIVERS="i915 i965 r300 r600 radeonsi nvidia"
    GRAPHIC_DRIVERS="r300 r600 radeonsi i915 i965 vmware virtio"
```

### CODE REVIEW

File: packages/wayland/weston/package.mk [hunk 1]
```diff
@@ -2,43 +2,46 @@
...
-    cp $PKG_DIR/config/weston.ini $INSTALL/usr/share/weston
+  mkdir -p ${INSTALL}/usr/share/weston
+    cp ${PKG_DIR}/config/weston.ini ${INSTALL}/usr/share/weston
 
-  rm -r $INSTALL/usr/share/wayland-sessions
-  rm -r $INSTALL/usr/lib/weston-simple-im
+  rm -r ${INSTALL}/usr/share/wayland-sessions
+  rm -r ${INSTALL}/usr/bin/weston-calibrator 
+  rm -r ${INSTALL}/usr/bin/weston-simple-*
+  rm -r ${INSTALL}/usr/bin/weston-touch-calibrator
```
Comment (MilhouseVH) [inline packages/wayland/weston/package.mk]: You could use `safe_remove` instead.

**Comment by MilhouseVH** [comment]
LGTM (other than small nit).

Are there any other "light-weight, simple, fast" XML parsers that we haven't yet included in this distribution? `pugixml` might be the 4th, or possibly 5th... :(

**Comment by chewitt** [comment]
https://xkcd.com/927/ :)

**Comment by 5schatten** [comment]
@CvH -> updated

File: packages/mediacenter/kodi/patches/kodi-100.14-use-alsa-and-pulse-together.patch [hunk 2]
```diff
@@ -1,17 +1,18 @@
-From 0bb21ebfe5f26545ccc9f105560c6c0fcbce77b3 Mon Sep 17 00:00:00 2001
-From: MilhouseVH <milhouseVH.github@nmacleod.com>
-Date: Wed, 26 Sep 2018 02:47:03 +0100
-Subject: [PATCH] allow using alsa and pulse together
+From 2e7de29a69b37ad0a7d8632c5aafcdd02a4f550d Mon Sep 17 00:00:00 2001
+From: 5schatten <supervisedthinking@gmail.com>
+Date: Tue, 7 May 2019 01:30:10 +0200
+Subject: [PATCH] [PATCH] allow using alsa and pulse together
```
Comment (CvH) [inline packages/mediacenter/kodi/patches/kodi-100.14-use-alsa-and-pulse-together.patch]: `Subject: [PATCH] allow using alsa and pulse together` :)

Comment (5schatten) [reply packages/mediacenter/kodi/patches/kodi-100.14-use-alsa-and-pulse-together.patch]: fixed

**Comment by MilhouseVH** [comment]
If this is needed for libreelec-9.2 can we create a backport PR? I'm maintaining #3488 for trivial backports but these Wayland changes may warrant their own backport PR!

**Comment by 5schatten** [comment]
I don't think X11 will be dropped in LE 9.2 so I guess it's a _nice to have_ for this point release but it won't be necessary.

**Pull request merged**